sprint-1/step-2

rsirazhdinov opened the pull request.

Commit:
sprint-1/step-2

SergeyK2004 commented on src/components/burger-category/burger-category.jsx:
```diff
@@ -0,0 +1,31 @@
+                    onClick={handleOpenModal} 
+                    count={1}
+                    className='ml-6' 
+                    key={ingredient._id}
```
Отлично:
Хорошо, что используете id для ключа

SergeyK2004 commented on src/components/burger-constructor/burger.constructor.jsx:
```diff
@@ -0,0 +1,82 @@
+            <ul className={styles.burger_constructor_box}>
+                {ingredients.map((item, i) => {
+                    return (
+                        <li key={i} className={styles.element_row_box}>
```
Использование индекса элемента в качестве ключа недопустимо. необходимо использовать уникальный идентификатор. Для ингредиентов в конструкторе вам нужно будет его добавлять дополнительно, так как _id элемента не будет уникальным, в конструткор можно добавлять один и тот-же ингредиент дважды. Учтите это при дальнейшей работе над проектом.

SergeyK2004 commented on src/components/ingredient-details/ingredient-details.jsx:
```diff
@@ -0,0 +1,52 @@
+export default function IngredientDetails({ingredient}) {
+    return (
+        <div className={ingredientDetailsStyles.modal_content}>
+            <img className='mb-4' src={ingredient.image_large} alt=''/>
```
Надо исправить:
Каждому элементу img нужно указать альтернативный текст для озвучивания специальными программами. Значение alt должно быть информативным. Подробнее про использование альтернативного текста на изображениях можно прочитать здесь https://habr.com/ru/articles/341810/

SergeyK2004 commented on src/components/ingredient/burger-ingredient.jsx:
```diff
@@ -0,0 +1,27 @@
+    return (
+        <article className={styles.container} onClick={() => onClick(ingredient)}>
+            {count > 0 && <Counter count={count} size="default" extraClass="m-1"/>}
+            <img className={`ml-4 mr-4 mb-1 ${styles.img}`} src={image} alt="Картинка ингредиента"/>
```
Надо исправить:
Значение `alt` должно быть информативным. используйте название ингредиента.

Исправьте во всем приложении.

SergeyK2004 replied:
Надо исправить:
Значение `alt` должно быть информативным. используйте название ингредиента.

Исправьте во всем приложении.

SergeyK2004 commented on src/utils/burger-api.js:
```diff
@@ -0,0 +1,16 @@
+const API_URL = 'https://norma.nomoreparties.space/api';
+
+const checkResponse = (res) => {
+    return res.ok ? res.json(): res.json().then((err) => Promise.reject(err))
```
Хорошо, что проверяете ответ сервера на `ok`

SergeyK2004 commented on src/utils/burger-types.jsx:
```diff
@@ -0,0 +1,18 @@
+import PropTypes from 'prop-types'
+
+export const ingredientType = {
```
Здорово, что вынесли описание типа ингредиента в отдельный файл.

SergeyK2004 commented on src/components/menu-item/menu-item.jsx:
```diff
@@ -0,0 +1,38 @@
+export default function MenuItem({iconName, iconType, title, isActive}) {
+    return (
+
+        <div className={`p-5 mb-4 mt-4 ml-2 ${menuItemStyles.menu_item}`}><span className={menuItemStyles.icon}>{getIconFromName(iconName, iconType)}</span> <p
```
Можно лучше:
Пункты меню лучше реализовать через тег <a>, так как это ссылки.

SergeyK2004 commented on src/components/burger-ingredients/burger-ingredients.jsx:
```diff
@@ -0,0 +1,53 @@
+            <p className=" mt-10 text text_type_main-large">
+                Соберите бургер
+            </p>
+            <nav style={{display: 'flex'}} className="mt-5">
```
Надо исправить:
Для стилизации компонентов необходимо использовать CSS-модули.

SergeyK2004 commented on src/components/app/app.jsx:
```diff
@@ -0,0 +1,75 @@
+                    alert("Во время загрузки произошла ошибка");
+                }
+            )
+    }, [])
```
Хорошо, что использовали пустой массив зависимостей.

SergeyK2004 reviewed (changes requested):
Здравствуйте. 

Вы проделали большую работу, что мне понравилось:

Хорошо выполнена верстка
Тип ингредиента вынесен в отдельный файл, чтобы не дублировать его в разных компонентах
Использование пустого массива зависимостей [] в useEffect 
Использование атрибута key, и здорово, что для него вы используете уникальный _id, а не индекс массива в списке ингредиентов.

но есть некоторые недочеты:

* https://disk.yandex.ru/d/WWwv_q7FjZHL5w Сайт должен занимать все пространство страницы по вертикали.
* Модальные окна должны закрываться при клике в оверлей.
* Вам нужно удалить лишние файлы из проекта https://disk.yandex.ru/d/C_1uBjAWRb8QxA
* https://disk.yandex.ru/d/h-vbZbvvIFKxUg Булка не должна быть в составе бургера, булка отображается только вверху и внизу в отдельных компонентах.
* Каждой картинке (тегу img) нужен информативный атрибут alt
* Инлайн-стилей style={{не должно быть в проекте, если они не меняются динамически в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных .module.css файлах, а в коде компонентов используются модульные css-стили.

Исправьте указанные недочеты и работа будет принята.

Commit:
sprint-1/step-2
поправлено следующее:
-https://disk.yandex.ru/d/WWwv_q7FjZHL5w Сайт должен занимать все пространство страницы по вертикали.
-Модальные окна должны закрываться при клике в оверлей.
-Вам нужно удалить лишние файлы из проекта https://disk.yandex.ru/d/C_1uBjAWRb8QxA
-https://disk.yandex.ru/d/h-vbZbvvIFKxUg Булка не должна быть в составе бургера, булка отображается только вверху и внизу в отдельных компонентах.
-Каждой картинке (тегу img) нужен информативный атрибут alt
-Инлайн-стилей style={{не должно быть в проекте, если они не меняются динамически в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных .module.css файлах, а в коде компонентов используются модульные css-стили.

SergeyK2004 reviewed (changes requested):
Очень хорошо.
Часть недочетов устранена, но еще часть осталась. Для того, чтобы работа была принята, необходимо устранить все недочеты.

* https://disk.yandex.ru/d/NVf3yclbSvRRdg Сайт по вертикали должен занимать все доступное пространство.
* https://disk.yandex.ru/d/K6yYlSWotNh-nw Вам нужно удалить лишние файлы из проекта 
* Модальные окна должны закрываться при клике в оверлей, но не в сам контент окна. Т.е. если открыто модальное окно описания ингредиента, то клик в картинку ингредиента не должен закрывать модалку. Тольк клик в оверлей (за пределами самого окна) закрывает модальное окно
* Каждой картинке (тегу img) нужен информативный атрибут alt

Commit:
sprint-1/step-2
поправлено следующее:
https://disk.yandex.ru/d/WWwv_q7FjZHL5w Сайт должен занимать все пространство страницы по вертикали.
Модальные окна должны закрываться при клике в оверлей.
Вам нужно удалить лишние файлы из проекта https://disk.yandex.ru/d/C_1uBjAWRb8QxA
https://disk.yandex.ru/d/h-vbZbvvIFKxUg Булка не должна быть в составе бургера, булка отображается только вверху и внизу в отдельных компонентах.
Каждой картинке (тегу img) нужен информативный атрибут alt
Инлайн-стилей style={{не должно быть в проекте, если они не меняются динамически в коде компонента. Это затрудняет управление стилями. Все стили описываются в модульных .module.css файлах, а в коде компонентов используются модульные css-стили.

SergeyK2004 reviewed (approved):
Отличная работа.
Замечаний больше нет. Ваш проект принят.
Желаю успехов в дальнейшем обучении.
